File: fixtures/python/simple_function.py
from pathlib import Path
import json


def read_config(path: str) -> dict:
    """Read and parse a JSON configuration file."""
    with open(path) as f:
        data = json.load(f)
    validated = validate_config(data)
    return validated


def validate_config(data: dict) -> dict:
    required_keys = ["name", "version", "entries"]
    for key in required_keys:
        if key not in data:
            raise ValueError(f"Missing required key: {key}")
    if not isinstance(data["entries"], list):
        raise TypeError("entries must be a list")
    return data

File: fixtures/typescript/simple_function.ts
import { readFileSync } from 'fs';
import path from 'path';

interface Config {
    name: string;
    version: string;
    entries: ConfigEntry[];
}

interface ConfigEntry {
    id: number;
    label: string;
    enabled: boolean;
}

type ConfigValidator = (config: Config) => boolean;

export function readConfig(filePath: string): Config {
    const fullPath = path.resolve(filePath);
    const raw = readFileSync(fullPath, 'utf-8');
    const parsed: Config = JSON.parse(raw);
    if (!validateConfig(parsed)) {
        throw new Error('Invalid configuration');
    }
    return parsed;
}

function validateConfig(config: Config): boolean {
    if (!config.name || !config.version) {
        return false;
    }
    if (!Array.isArray(config.entries)) {
        return false;
    }
    return config.entries.every(e => typeof e.id === 'number');
}
